backend/Tournament/managers.py

```diff
@@ -0,0 +1,174 @@
+from django.db import transaction
+from django.utils import timezone
+from datetime import timedelta
+
+from .models import Tournament, \
+                        TournamentPlayer
+from .serializers import TournamentCreationSerializer
+from .exceptions import TournamentSetupException, \
+                            TournamentInProgressException
+from .tournamentState import TournamentState
+from .tournamentMacros import TOURNAMENT_EXPIRY_TIME_SECONDS
+from User.models import User
+from Match.models import Match
+from Match.matchState import MatchState
+from Player.models import Player
+
+import sys
+
+class TournamentSetupManager:
+    @staticmethod
+    def create_tournament_and_tournament_player_for_host(validated_data: TournamentCreationSerializer.validated_data):
+        try:
+            with transaction.atomic():
+                tournament = Tournament.objects.create(
```

Comment:
I think the creation of the tournament should be in the tournamentmanager and the creation of tournamentplayer in tournamentplayermanager. Then there would be some middle manager that will call both of these managers to create the two.